flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -785,15 +783,6 @@ private ApplicationReport startAppMaster(
 		paths.add(remotePathJar);
 		classPathBuilder.append(flinkJarPath.getName()).append(File.pathSeparator);
 
-		// set the right configuration values for the TaskManager
-		configuration.setInteger(
-				TaskManagerOptions.NUM_TASK_SLOTS,
-				clusterSpecification.getSlotsPerTaskManager());
-
-		configuration.set(
-				TaskManagerOptions.TOTAL_PROCESS_MEMORY,
-				MemorySize.parse(clusterSpecification.getTaskManagerMemoryMB() + "m"));
```

Comment:
I think we could change `YarnClusterDecriptor#validateClusterResources` to return a `ClusterSpecification` with the `taskManagerMemoryMB` of the passed in `ClusterSpecification`.

Reply:
Then I believe the following codes in `YarnClusterDescriptor#validateClusterResources` can also be removed.
```
if (taskManagerMemoryMb < yarnMinAllocationMB) {
	taskManagerMemoryMb =  yarnMinAllocationMB;
}
```